explorer/ast/value.h

```diff
@@ -1466,7 +1467,7 @@ class ParameterizedEntityName : public Value {
 // These values are used to represent the second operand of a compound member
 // access expression: `x.(A.B)`, and can also be the value of an alias
 // declaration, but cannot be used in most other contexts.
-class MemberName : public Value {
+class MemberName : public Value, Printable<MemberName> {
```

Comment:
Do you need the base class here? I'd expect the printing functionality to get inherited from the base class `Value`.

Reply:
Note it's not virtual:
```
void Print(llvm::raw_ostream& out) const { member_->Print(out); }
```

Reply:
Oh no, and the base class version does something different from the derived class version too! Yuck!

Reply:
#3168